packages/evm/src/__test__/fungible.test.ts

```diff
@@ -164,13 +164,41 @@ describe('Fungible - Fee', () => {
       calculateFee: jest.fn().mockResolvedValue([BigNumber.from(0)]),
     });
 
-    const transfer = await createEvmFungibleAssetTransfer(TRANSFER_PARAMS);
+    const transfer = await createFungibleAssetTransfer(TRANSFER_PARAMS);
     const fee = await transfer.getFee();
 
     // expect(fee.fee).toEqual(0n);
     expect(fee.type).toEqual('percentage');
     expect(fee.handlerAddress).toEqual('0x98729c03c4D5e820F5e8c45558ae07aE63F97461');
   });
+
+  it('should query fee with deposit data', async () => {
+    const calculateFee = jest.fn().mockResolvedValue([BigNumber.from(0)]);
+
+    (PercentageERC20FeeHandler__factory.connect as jest.Mock).mockReturnValue({
+      feeHandlerType: jest.fn().mockResolvedValue('percentage'),
+      calculateFee,
+      _resourceIDToFeeBounds: jest.fn().mockResolvedValue({
+        lowerBound: parseEther('10'),
+        upperBound: parseEther('100'),
+      }),
+      _domainResourceIDToFee: jest.fn().mockResolvedValue(BigNumber.from(100)),
+      HUNDRED_PERCENT: jest.fn().mockResolvedValue(10000),
+    });
+
+    (BasicFeeHandler__factory.connect as jest.Mock).mockReturnValue({
+      feeHandlerType: jest.fn().mockResolvedValue('percentage'),
+      calculateFee: jest.fn().mockResolvedValue([BigNumber.from(0)]),
+    });
+
+    const transfer = await createFungibleAssetTransfer(TRANSFER_PARAMS);
+    await transfer.getFee();
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    expect(calculateFee.mock.calls[0][4]).toEqual(
```

Comment:
it's better to use const to show what is the expected value f.e  
```
const expectedFeeHandlerId = calculateFee.mock.calls[0][4]
```
or whatever this long string means 😅